resources.go

```diff
@@ -843,7 +855,21 @@ func Provider() tfbridge.ProviderInfo {
 			},
 
 			// Cloud IoT Core resources
-			"google_cloudiot_registry": {Tok: gcpResource(gcpKMS, "Registry")},
+			"google_cloudiot_registry": {
+				Tok: gcpResource(gcpKMS, "Registry"),
+				Fields: map[string]*tfbridge.SchemaInfo{
+					// This property's nested type name conflicts with the nested type of the existing (now deprecated)
+					// `event_notification_config` property (singular, a TypeMap). A conflict occurs because the new
+					// `event_notification_configs` property (plural, a TypeList) is a TypeList, which we singularize.
+					// To avoid the conflict, we override the nested type name for the new property, appending an "Item"
+					// suffix.
+					"event_notification_configs": {
+						Elem: &tfbridge.SchemaInfo{
+							NestedType: "RegistryEventNotificationConfigItem",
```

Comment:
@lukehoban, just to confirm, are you OK with using the `Item` suffix for this? I considered other names like `Entry` and `Element` but didn't feel too strongly about any of them. Open to changing it or other ideas.

Using a suffix like this seems in alignment with the place you [pointed out](https://github.com/pulumi/pulumi-gcp/issues/212#issuecomment-528487118) where we used a `Collection` suffix.